geometry/tagged/difference.js

```diff
@@ -1,17 +1,56 @@
+import { getClosedGraphs, getGraphs } from './getGraphs.js';
+
+import { cutVolumeIncrementally } from '../graph/cutVolumeIncrementally.js';
+import { cutVolumeSingly } from '../graph/cutVolumeSingly.js';
+import { cutVolumeSinglyRecursive } from '../graph/cutVolumeSinglyRecursive.js';
 import { fill as fillOutlineGraph } from '../graph/fill.js';
 import { fromPaths as fromPathsToGraph } from '../graph/fromPaths.js';
 import { getFaceablePaths } from './getFaceablePaths.js';
-import { getGraphs } from './getGraphs.js';
 import { difference as graphDifference } from '../graph/difference.js';
 import { rewrite } from './visit.js';
 import { toConcreteGeometry } from './toConcreteGeometry.js';
 
-export const difference = (geometry, ...geometries) => {
-  geometries = geometries.map(toConcreteGeometry);
+export const difference = (geometry, options = {}, ...geometries) => {
+  if (
+    !['incremental', 'single', 'single_recursive', 'basic', undefined].includes(
```

Comment:
So all of the modes are exposed now in the api which is nice for testing. Long term do you think all of them will be exposed or if there is a clear winner will just one be exposed in the end?